### PR TITLE
SDK to TypeScript

### PR DESCRIPTION
**Pull request by thomasmost**
What Changes
---
* Deprecates the KavaClient class (previous versions of the SDK will still support it; we can migrate this if necessary)
* Converts the message-based SDK to TypeScript

Testing
---
Test locally by running `npm i @kava-labs/javascript-sdk@beta`

### CODE REVIEW

File: src/msg/kava/index.ts [hunk 1]
```diff
@@ -154,111 +193,137 @@ function newMsgSubmitProposal(proposal, proposer, committeeID) {
...
     value: {
       proposal_id: String(proposalID),
       voter: voter,
-      vote_type: voteType
+      vote_type: voteType,
     },
   };
 }
 
+function newMsgVoteGovernance(proposalID: string, voter: string, voteType: VoteType) {
```
Comment (downing034) [inline src/msg/kava/index.ts:207]: do we want to move this on the the `src/msg/cosmos/index.ts` file since it's not kava specific?

Comment (thomasmost) [reply src/msg/kava/index.ts:207]: ah, sure

**Review by downing034 — APPROVED** [review]
Everything is looking snazzy. I added one nit comment about possibly moving a message for clarity of which "protocol" it's tied to.

My only open question here is if we're going to break some non webapp related things. 
 Seems we should be safe since we're both upgrading the version and setting it to a beta so others shouldn't have to upgrade if they don't want to, but just want to call it out.

**Comment by thomasmost** [comment]
> My only open question here is if we're going to break some non webapp related things.
> Seems we should be safe since we're both upgrading the version and setting it to a beta so others shouldn't have to upgrade if they don't want to, but just want to call it out.

Yeah, agreed, and I really don't think anyone's using the KavaClient since that's... already broken (at least in a couple places)

**Pull request merged**